src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseDataFileService.cs

```diff
@@ -543,5 +575,117 @@ private async Task DeleteBatchFiles(Guid releaseId, File dataFile)
                 );
             });
         }
+
+        private async Task<Guid> CreateReleaseAndSubjectHierarchy(Release release, string subjectFilename,
```

Comment:
As mentioned in Slack, would refactor this into a repository i.e. `ReleaseRepository`. After doing this, I would suggest updating it to have the following method signature: `CreateStatisticsReleaseAndSubjectHierarchy(Guid releaseId, string subjectFilename, string subjectName)`. 

It's not a great idea to be passing around raw entities through different services as we have no guarantees they are hydrated correctly (as you've already thought about with your null check).